### PR TITLE
Nc news 5 

### PR DESCRIPTION
**Pull request by royr5**
add GET /api/articles endpoint

### CODE REVIEW

File: __tests__/index.test.js [hunk 1]
```diff
@@ -49,6 +50,34 @@ describe("/api/topics", () => {
 });
 
 describe("/api/articles", () => {
+  test("GET:200 sends an array of articles objects", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then((response) => {
+        const arr = response.body.articles;
+        expect(arr.length).toBe(5);
```
Comment (johnhutcheon) [inline __tests__/index.test.js:59]: Nice work checking the length before forEach 👍

File: __tests__/index.test.js [hunk 2]
```diff
@@ -49,6 +50,34 @@ describe("/api/topics", () => {
...
+        expect(arr).toBeSortedBy("created_at", { descending: true });
```
Comment (johnhutcheon) [inline __tests__/index.test.js:60]: Well done with this line too! 🙌

File: __tests__/index.test.js [hunk 3]
```diff
@@ -49,6 +50,34 @@ describe("/api/topics", () => {
...
+          expect(typeof article.topic).toBe("string");
+          expect(typeof article.author).toBe("string");
+          expect(typeof article.created_at).toBe("string");
+          expect(typeof article.votes).toBe("number");
+          expect(typeof article.article_img_url).toBe("string");
+          expect(typeof article.comment_count).toBe("string");
+        });
+      });
+  });
+  test("404: respond with an error message saying path is not found", () => {
```
Comment (johnhutcheon) [inline __tests__/index.test.js]: This test isn't specific to api/articles. It should either be in it's own describe block or removed.

File: app.js [hunk 4]
```diff
@@ -11,6 +11,7 @@ const {
 app.get("/api/topics", getTopics);
 app.get("/api", getEndpoints);
 
+app.get("/api/articles", getArticles);
 app.get("/api/articles/:article_id", getArticles);
```
Comment (johnhutcheon) [inline app.js]: We shouldn't be re-using controllers. `getArticles` should only be the controller for `api/articles`. You should have a separate controller for `/api/articles/:article_id`, preferably called something like `getSingleArticle` 👍

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Hey Roshan, just a few changes need to be made here please...
- Each endpoint should have it's own controller and model. We shouldn't be re-using them for different endpoints.
Please refactor and resubmit. 👍

**Review by nickkagame — APPROVED** [review]
Nice work Roshan

**Pull request merged**